NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -12,35 +12,200 @@
-
     Bx, Bz = corsika['inputs'].attrs["MAGNET"]
     B_inclination = np.arctan2(Bz, Bx)
 
     B_strength = (Bx ** 2 + Bz ** 2) ** 0.5 * units.micro * units.tesla
-
     # in local coordinates north is + 90 deg
     magnetic_field_vector = B_strength * hp.spherical_to_cartesian(np.pi * 0.5 + B_inclination, 0 + np.pi * 0.5)
-
     return zenith, azimuth, magnetic_field_vector
 
+def get_geomagnetic_angle(zenith, azimuth, magnetic_field_vector):
+    """
+    Calculates the angle between the geomagnetic field and the shower axis.
+
+    Parameters
+    ----------
+    zenith : float
+        zenith angle in radians
+    azimuth : float
+        azimuth angle in radians
+    magnetic_field_vector : np.array (3)
+        magnetic field vector in cartesian coordinates
+
+    Returns
+    -------
+    geomagnetic_angle : float
+        geomagnetic angle in radians
+    """
+    shower_axis_vector = hp.spherical_to_cartesian(zenith, azimuth)
+    geomagnetic_angle = hp.get_angle(magnetic_field_vector, shower_axis_vector)
+    return geomagnetic_angle
+
+def coreas_observer_to_nuradio_efield(corsika, out_dir=None, save_dict=False):
+    """
+    converts the electric field from the corsika file to NuRadio units
+
+    Parameters
+    ----------
+    corsika : hdf5 file object
+        the open hdf5 file object of the corsika hdf5 file
+    outdir: str
+        Directory to save the resulting dictionary. Default is None. Outputname is based on hdf5_file
+    save_dict: bool
+        Whether to save the resulting dictionary to the specified output file. Default is False.
+
+    Returns
+    -------
+    efield: np.array (3, n_samples)
+        efield with three polarizations (r, theta, phi)
+    efield_times: np.array (n_samples)
+    """
+    n_observer = len(corsika["CoREAS"]['observers'].items())
+    group = (corsika["CoREAS"]['observers'])
+    keys = list(group.keys())
+    n_trace_samples = len(corsika["CoREAS"]['observers'][keys[0]][:,0])
+
+    zenith, azimuth, magnetic_field_vector = get_angles(corsika)
+    cs = coordinatesystems.cstrafo(zenith, azimuth, magnetic_field_vector)
+    
+    electric_field = np.zeros((n_observer, n_trace_samples, 4))
+
+    obs_positions = []
+    electric_field_on_sky = []
+    for j_obs, observer in enumerate(corsika['CoREAS']['observers'].values()):
+        #account for different coordinate systems (see get_angles function)
+        obs_positions.append(np.array([-observer.attrs['position'][1], observer.attrs['position'][0], 0]) * units.cm)
+
+        efield = np.array([observer[()][:,0]*units.second,
+                            -observer[()][:,2]*conversion_fieldstrength_cgs_to_SI,
+                            observer[()][:,1]*conversion_fieldstrength_cgs_to_SI,
+                            observer[()][:,3]*conversion_fieldstrength_cgs_to_SI])
+
+        efield_geo = cs.transform_from_magnetic_to_geographic(efield[1:,:])
```

Comment:
This operation is done many times in this file, but also in any other analysis using CoREAS in NRR module. So I think it would make sense to put this part into its own function.

Reply:
I agree, that should not be reimplemented here. Why don't you use the function that is already provided in coreas.py? All general functions go into that file

Reply:
I used it here because I wanted observer position and efield in one loop instead of two. It changed for the new PR